src/pydata_sphinx_theme/assets/scripts/pydata-sphinx-theme.js

```diff
@@ -392,6 +384,70 @@ if (versionSwitcherBtns.length) {
   });
 }
 
+/*******************************************************************************
+ * Warning banner when viewing non-stable version of the docs.
+ */
+
+/**
+ * Show a warning banner when viewing a non-stable version of the docs.
+ *
+ * adapted 2023-06 from https://mne.tools/versionwarning.js, which was
+ * originally adapted 2020-05 from https://scikit-learn.org/versionwarning.js
+ *
+ * @param {Array} data The version data used to populate the switcher menu.
+ */
+function showVersionWarningBanner(data) {
+  const version = DOCUMENTATION_OPTIONS.VERSION;
+  // figure out what latest stable version is
+  var preferredEntries = data.filter((entry) => entry.preferred);
+  if (preferredEntries.length !== 1) {
+    const howMany = preferredEntries.length == 0 ? "No" : "Multiple";
+    throw new Error(
+      `[PST] ${howMany} versions marked "preferred" found in versions JSON`
+    );
+  }
+  const preferredVersion = preferredEntries[0].version;
+  const preferredURL = preferredEntries[0].url;
+  // if already on preferred version, nothing to do
+  const versionsAreComparable = validate(version) && validate(preferredVersion);
+  if (versionsAreComparable && compare(version, preferredVersion, "=")) {
+    return;
+  }
+  // now construct the warning banner
+  var outer = document.createElement("div");
+  const middle = document.createElement("div");
+  const inner = document.createElement("div");
+  const bold = document.createElement("strong");
+  const button = document.createElement("a");
+  // these classes exist since pydata-sphinx-theme v0.10.0
+  outer.classList = "bd-header-version-warning container-fluid";
+  middle.classList = "bd-header-announcement__content";
+  inner.classList = "sidebar-message";
+  button.classList =
+    "sd-btn sd-btn-danger sd-shadow-sm sd-text-wrap font-weight-bold ms-3 my-1 align-baseline";
+  button.href = `${preferredURL}${DOCUMENTATION_OPTIONS.pagename}.html`;
+  button.innerText = "Switch to latest stable version";
+  button.onclick = checkPageExistsAndRedirect;
+  // add the version-dependent text
+  inner.innerText = "This is documentation for an ";
+  const isDev = version.includes("dev");
```

Comment:
I presume this check is broad enough, but depending on how projects work perhaps it would be worth checking for "pre", "rc", "beta" etc.?

Reply:
good catch!  I thought the node `compare-versions` library would handle those but it turns out it only handles "beta":

```js
> cv.validate("1.24.0-beta1");
true
> cv.validate("v0.15.2pre"); //pandas
false
> cv.validate("1.24.0rc2"); //numpy
false
```

I'll push a fix